### PR TITLE
MM-22038: removing deactivate users from open private channels without private messages with me

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Removing deactivate users from open private channels when the users are deactivated users and we don't have a private channel between us.

This is because only makes sense to open a private message with a deactivated user to read a previous conversation.

#### Ticket Link
[MM-22038](https://mattermost.atlassian.net/browse/MM-22038)

### CODE REVIEW

File: components/more_direct_channels/more_direct_channels.test.jsx [hunk 1]
```diff
@@ -243,4 +244,48 @@ describe('components/MoreDirectChannels', () => {
             done();
         });
     });
+
+    test('should excluded deleted users if there is not direct channel between users', () => {
```
Comment (fm2munsh) [inline components/more_direct_channels/more_direct_channels.test.jsx]:
```suggestion
    test('should exclude deleted users if there is not direct channel between users', () => {
```

File: components/more_direct_channels/__snapshots__/more_direct_channels.test.jsx.snap [hunk 2]
```diff
@@ -1,5 +1,172 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`components/MoreDirectChannels should excluded deleted users if there is not direct channel between users 1`] = `
```
Comment (fm2munsh) [inline components/more_direct_channels/__snapshots__/more_direct_channels.test.jsx.snap]:
```suggestion
exports[`components/MoreDirectChannels should exclude deleted users if there is not direct channel between users 1`] = `
```

**Comment by amyblais** [comment]
I decided to push this to v5.21 because this is not a new regression and I don't how risky this is. Let me know if this is a more urgent issue,

**Pull request merged**